calyptos/plugins/validator/topology.py

```diff
@@ -9,8 +9,9 @@ def validate(self):
 
         self.topology = self.environment['default_attributes']['eucalyptus']['topology']
         assert self.roles['clc']
-        assert self.roles['walrus']
         assert self.roles['user-facing']
+        if self.roles['walrus']:
+            assert self.roles['walrus']
```

Comment:
this is redundant and isnt actually checking anything anymore. Basically saying, "if this exists, make sure it exists"